docs/docs/09-rfcs/2023-03-08-wing-preview-environment.md

```diff
@@ -0,0 +1,278 @@
+| --------------- | ---------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------- | -------------------- |
+| main.w          | ✅ Ready ([logs](https://wing.cloud/gh-account/gh-repo/gh-branch/main/logs/))            | [Visit Preview](https:/gh-repository-gh-branch-3312-main.wing.cloud.app)         | Jul 31, 2023 8:01am  |
+| another.main.w  | ✅ Ready ([logs](https://wing.cloud/gh-account/gh-repo/gh-branch/another-main/logs/))    | [Visit Preview](https:/gh-repository-gh-branch-3312-another-main.wing.cloud.app) | Jul 30, 2023 10:01am |
+| failed.main.w   | ❌ Failed ([logs](https://wing.cloud/gh-account/gh-repo/gh-branch/entry-file/logs/))     |                                                                                  | Jul 31, 2023 8:01am  |
+| building.main.w | 🔄 Building ([logs](https://wing.cloud/gh-account/gh-repo/gh-branch/entry-file/logs/))   |                                                                                  | Jul 31, 2023 8:01am  |
+| stale.main.w    | ⏸️ Stale ([lean more](https://wing.cloud/gh-account/gh-repo/gh-branch/entry-file/logs/)) |                                                                                  | Jul 31, 2023 8:01am  |
+
+TBD - tests results per entry point
+
+**Disclaimer**: Developers that have write access and can create PRs but didn't signed-up to Wing Cloud will not be able to create preview environments.
+In this case the above comment will contain a direct link for signing up to Wing Cloud.
+After signing up, the developer will be able to create a preview environment for the PR and one will be created automatically after the next code change.
+
+#### Multiple Entry Points
+
+A developer can configure multiple entry points in a single PR.
+Every `*.main.w` file in the repository will be considered as an entry point and will have a dedicated preview environment.
+All preview environments will be available in the PR comment and will be updated upon code changes.
+
+#### Populate Environment with Initial Data
+
+Out of scope for this project.
+
+#### Automatically Running Tests
+
+Upon PR creation and code changes, Wing Cloud Preview Environments will automatically run all tests defined for each entry point.
+For each test run, a new simulator will be created and will be destroyed upon test completion.
+The test results will be available in the PR comment for each entry point.
+
+#### Running Tests Manually
+
+Running tests on your environment is easy using the Wing Console tests UI.
+Each test will reload the simulator to create a clean environment.
+
+Once the test is over it will **not** clean up the simulator data, so you can debug the environment and share outputs with your teammates.
+
+#### User-Controlled Environments termination
+
+Closing a PR will terminate the corresponding preview environments.
```

Comment:
I really think we'll need some mechanism for suspending stale PE's. If there's important state we might need to store it before suspending. But just spinning up a container per PR means that at some point we'll have lots of containers running forever.